pkg/proxy/proxy.go

```diff
@@ -130,7 +130,7 @@ func (k *Currency) sendTransaction(ticker string, txHex string) error {
 	bodyReader := bytes.NewReader(postRequest.Body)
 
 	// create an http request
-	httpReq, err := http.NewRequest("POST", k.rpcURL, bodyReader)
+	httpReq, err := http.NewRequest("POST", postRequest.URL, bodyReader)
```

Comment:
Is `postRequest.URL` necessarily different from `k.rpcURL`?

Reply:
Yes, depending on which chain. For example the cosmos chain updates the URL while the ethereum chain does nothing to the URL.

The idea is that each chain might have a different URL path. The value of `k.rpcURL` comes directly from `currency.toml`. This works when the root endpoint is all you need but in cosmos the this gets appended to the root endpoint:

```go
URL := fmt.Sprintf("%s/broadcast_tx_async?tx=0x%s", rpcURL, txHex)
```

The other solution is to put the following in the `currency.toml` file:
```toml
Ticker = "tbnb"
RPCUser = "rpcuser"
RPCPass = "rpcpassword"
RPCURL = "https://tbinance.hashcloak.com/broadcast_tx_async?tx=0x" <- This
LogDir = "/conf"
LogLevel = "DEBUG"
```

Reply:
If we update this to a pointer then we can avoid the duplicate variable.

Reply:
I see. What I had in mind was "Couldn't you just set the URL in the `currency.toml` file?" But you can't, as you say, since when you generate the raw transaction it needs to be appended in a `POST` request.